main/webapp/modules/core/scripts/util/dialog.js

```diff
@@ -42,6 +42,13 @@ var escapeKey = function(event) {
   }
 }
 
+/**
+ * Show a dialog.
+ * @public
+ * @param {jQuery} elmt The dialog element to show.
```

Comment:
This is neat but confusing me.  Does this mean that it takes a jQuery object?  Aren't those simply regular JS objects?  Why not `{object}` ?  Sorry but learning here and wondering if my VSCode could make sense of this better with the right extension...maybe that's my problem here?  Any tips @Abbe98 ?  How's it look for you in VSCode when you hover over that "jQuery"?

Reply:
This is a great question. The param types are not necessarily native JS types, they can be anything. These can be declared either with an `@external` annotation as mentioned by @SoryRawyer or with `@typedef`. Some very common "types" like jQuery's `$` and DOM "types" like HTMLElement tend to, however, be left alone(some editors/extensions also bundle common `@typedef` declarations).

I'm not very opinionated about how it's done as long as they are resonable named.